docs/source/troubleshooting/logs.rst

```diff
@@ -0,0 +1,153 @@
+
+-  Per the default ``logrotate.conf`` file, logs are rotated daily with up to 4 copies archived.
+   Older archives are deleted when a log is rotated if the maximum is hit.
+   This configuration is overwritten by the configurations for individual files.
+
+   -  For example, a configuration file might instead specify that the log is rotated hourly instead of the default setting.
+
+When modifying these settings, keep in mind that many of the logs are stored in */var/log* and are therefore stored in volatile memory by default.
+If the modified settings result in logs growing too large, it can result in the system running out of memory.
+Even for logs stored in the device’s physical storage, it’s important to ensure that the logs don’t accidentally file up the device storage completely.
+Use reasonable settings and consider rotating logs that grow quickly at specific file sizes whenever using non-default settings.
+
+Logging to the System Log(s)
+============================
+
+When logging status or event information, it may be desirable to write to one of the logs outlined above.
+The following list outlines some of the common logs and what methods are recommended for writing to them.
+
+-  ``/var/local/natinst/log/errlog.txt``
+
+   -  `Write to System Log VI <http://zone.ni.com/reference/en-XX/help/371361R-01/glang/write_to_system_log/>`
+      - On Linux Real-Time Systems, this will write to the ``errlog.txt`` file.
+
+   -  `RT Debug String VI <http://zone.ni.com/reference/en-XX/help/370715P-01/lvrtvihelp/rt_debug_strings/>`
+
+      - On Linux Real-Time Systems, when the **Interface** is set to **Write to system log**, the input string will be written to the *errlog.txt* file.
+
+-  ``/var/log/messages``
+
+   -  `System Exec VI <https://zone.ni.com/reference/en-XX/help/371361R-01/glang/system_exec/>`
+
+      - Using the command ``logger <string>`` with the System Exec VI will write that string to the ``messages`` log.
+
+Note that when writing to these logs that it’s important to be aware of how they behave.
+For example, by default the ``errlog.txt`` file will persist across reboots but be rotated by ``logrotate`` very quickly due to file size while the ``messages`` log will not persist across reboots but will be rotated less often.
+
+Enabling Persistent Logs
+========================
+By default, logs are stored in ``/var/log`` which is a symlink to ``/var/volatile/log``.
+The ``/var/volatile`` location is a RAM-based tmpfs filesystem whose contents are lost on each reboot.
+This section explains how to make those logs persistent.
+
+**Warning:** Performing the steps in this section can impact system performance.
+Normally these logs are stored in RAM, which is much quicker to read and write.
+By moving these logs to persistent storage, the CPU and disk performance may be impacted.
+Additionally, it is recommended that these steps only be used on PXI Linux Real-Time or x64 cRIOs unless specifically needed for troubleshooting.
```

Comment:
That's the case on both ARM and x64 target, even though the lifetime on the x64s is much greater.  I think you can just say "may impact the disk lifetime" or something like that?

Reply:
#22